bpf/bpf_lxc.c

```diff
@@ -238,12 +238,18 @@ static inline int ipv6_l3_from_lxc(struct __sk_buff *skb,
 
 		return ipv6_local_delivery(skb, l3_off, l4_off, SECLABEL, ip6, tuple->nexthdr);
 	} else {
-#ifdef ENABLE_NAT46
-		/* FIXME: Derive from prefix constant */
-		if (unlikely((tuple->addr.p1 & 0xffff) == 0xadde)) {
-			do_nat46 = 1;
-			goto to_host;
-		}
+#if defined ENABLE_NAT46 && defined ENABLE_IPV4 && defined CONNTRACK
+		if (unlikely(isMappedIPv6Addr(&tuple->addr))) {
+			ret = ipv6_to_ipv4(skb, 14, htonl(LXC_IPV4));
```

Comment:
Are you sure the htonl() here is correct? `LXC_IPV4` should be in network byte order already.